Paaaaaain

comma opened the pull request:
<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->

Being unconscious makes you get no pain messages. Fxes #14267
Axed 'analgesic' var that was only used to take chem_effects[CE_PAINKILLER] in life()
CE_PAINKILLER effects now block pain/pain messages according to its severity rather than hardcoded values for all messages/pains.
Updated all usages of custom_pain to use power.
Shock effects are now not applied if mob is not awake. Fixes #14268 

feedback, suggestions, screams, insults?

comma committed:
Changed painkillers and pain handling a bit

Being unconscious makes you get no pain messages. Fxes #14267
Axed 'analgesic' var that was only used to take chem_effects[CE_PAINKILLER] in life()
CE_PAINKILLER effects now block pain/pain messages according to its severity rather than hardcoded values for all messages/pains.
Updated all usages of custom_pain to use power.
Shock effects are now not applied if mob is not awake. Fixes #14268

comma committed:
Merged pain() into handle_pain()

It was only place that used it

SkowronX commented:
False advertising, not enough bees or bee superpowers.

Kelenius commented:
Why are you nerfing painkillers?

Minijar commented:
> Why are you nerfing painkillers?

Because paracetamol should not block the pain of leg amputation or gunshots I think.

comma commented:
tramadol can go back t 80 sure, it doesn't really change much in pain message handling
Paracetamol was lowered because it'd let ignore you damage up to 25 for each limb, and that's a bit over the capacity imo.
It would let you ignore braindamge headaches and wounds up to 15 per limb still.
As for shock, paracetamol's value of 50 would let you ignore 20 brute damage AND a broken bone.
Now it'll make broken bone almost bearable as long as trauma is healed.

comma committed:
Changed tramadol back to 80

It was more or less a leftover of a change I done before I remembered about shock.

comma commented:
Pain message stuff is like this:
Up to 25 - ouchies that paracetamol can handle
Up to 80 - stuff like broken bones, non-deep surgery, shrapnel. Gonna need tramadol here.
Above 80 - currently only 80+ damage, amputations, internal organ surgery. Even tramadol is not enough to block out pain efficiently, gonna need OPycodone.

mwerezak commented on code/modules/mob/living/carbon/shock.dm:
```diff
@@ -14,25 +14,26 @@
+	if (!can_feel_pain())
+		src.traumatic_shock = 0
+		return 0
+	traumatic_shock = ..()
```
When taken together with line 22 and lines 31-35, won't this mean that painkillers will be unable to affect the traumatic_shock caused by broken bones, dislocations, and open organs?

mwerezak commented on code/modules/mob/living/carbon/shock.dm:
```diff
@@ -14,25 +14,26 @@
-		src.traumatic_shock = 0
+	traumatic_shock = max(0,traumatic_shock)
 
 	return src.traumatic_shock
```
Fix would be:

```
. = src.traumatic_shock
src.traumatic_shock = max(0, src.traumatic_shock)
```

Which would replace the return here and at line 36.
